### ISSUE TITLE
The Zarf Agent mutating webhook mutates objects on `CREATE` only, not `UPDATE`

### ISSUE THREAD

**Issue by RothAndrew** [defenseunicorns/zarf#564]
### Environment
Device and OS: EC2 instance running Ubuntu
App version: v0.19.5
Kubernetes distro being used: K3s v1.21.6+k3s1
Other:

### Steps to reproduce
1. Deploy a Flux-based setup that uses GitRepositories, where the `url:` param uses an "upstream" url that isn't "zarf-ified"
2. Update one of the GitRepositories in your config repo and push that new commit to gitea so that flux picks it up

### Expected result

The `url:` param still specifies a "zarf-ified" URL due to a mutation from the zarf agent

### Actual Result

The `url:` param now specifies the upstream "non-zarf-ified" URL

### Visual Proof (screenshots, videos, text, etc)

![image](https://user-images.githubusercontent.com/16000938/176030883-875a040a-09bb-431b-9d42-5e6a9d9847d5.png)

### Severity/Priority

Medium, this is gonna cause things to break in [DI2-ME](https://github.com/defenseunicorns/zarf-package-software-factory), but it won't imminently cause a production-down event

### Additional Context

The mutating webhook for [pods](https://github.com/defenseunicorns/zarf/blob/b1cf764340a8c0210435934f2543369d94c9f4b1/packages/zarf-agent/manifests/webhook.yaml#L36) as well as [GitRepositories](https://github.com/defenseunicorns/zarf/blob/b1cf764340a8c0210435934f2543369d94c9f4b1/packages/zarf-agent/manifests/webhook.yaml#L77) specify `CREATE` as the only operation when the agent will mutate the object. It should specify both `CREATE` and `UPDATE`.

**Comment by jeff-mccoy** [comment]
What's happening is we are advertising a binding with the webhook yaml, but the [actual code](https://github.com/defenseunicorns/zarf/blob/master/src/internal/agent/hooks/flux.go#L29) is only binding to `CREATE` currently.

**Comment by jeff-mccoy** [comment]
Won't be a big deal to fix it today, but I want to get the E2E to cover this as well and the test harness on master makes me really sad so going a little slow today with it (getting fixed in another PR)

**Comment by jeff-mccoy** [comment]
@YrrepNoj this might be a great one for you too look at too, to expand your understanding of the nuts-and-bolts.

**Comment by jeff-mccoy** [comment]
Haha no you're right that would be kind of useless, just trying to get a solid test foundation first while balancing blocking you on DI2ME

**Comment by jeff-mccoy** [comment]
I'll have to do a little testing too to see how the control plane passes objects on UPDATE vs CREATE.  I * THINK * they are the same..but need to test that.  Either way it's a pretty easy fix.

**Comment by RothAndrew** [comment]
Sounds good. I believe I'm unblocked for now by sticking with v0.19.5 and adding the zarf-ified URLs in, so that even if they mutate they are still exactly what they want them to be. I can then upgrade to the new version that fixes the issue and take back out the zarf-ified urls.

**Comment by RothAndrew** [comment]
@jeff-mccoy can we add this issue to the GA milestone?
